ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -316,8 +316,28 @@ extension SignalProducerType {
 	///
 	/// Returns a Disposable which can be used to interrupt the work associated
 	/// with the Signal, and prevent any future callbacks from being invoked.
-	public func start(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
-		return start(Event.sink(next: next, error: error, completed: completed, interrupted: interrupted))
+	public func startWithNext(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
```

Comment:
Hmm. The next-closure doesn't have to be optional but then the ordering will matter (e.g. `completed` must precede `next`), so that is not an option IMO. It does have to be the last argument to support the trailing closure.

I'm not sure. I think keeping it clean at the call site could be worth an somewhat un-intuitive argument ordering. This was the same style used with the previous overloading of `start`, but I agree it gets more confusing with the new `startWithNext` naming.

Looks like we have the following options:

1.
```swift 
producer.startWithNext({
    println("Next \($0)")
})

producer.startWithNext({
    println("Next \($0)")
  }, 
  error: {
    println("Error \($0)")
})
```

2.
```swift
producer.startWithNext {
    println("Next \($0)")
}

producer.startWithNext(
  next: {
    println("Next \($0)")
  }, 
  error: {
    println("Error \($0)")
})
```

3.
```swift 
producer.startWithNext {
    println("Next \($0)")
}

producer.startWithError {
    println("Error \($0)")
}
```